engine/schema/src/com/cloud/upgrade/dao/Upgrade481to490.java

```diff
@@ -53,6 +62,139 @@ public boolean supportsRollingUpgrade() {
+        try ( PreparedStatement updatePstmt = conn.prepareStatement(insertSql)) {
+            updatePstmt.executeUpdate();
+        } catch (SQLException ignored) {
+            s_logger.debug("Unable to insert mapping for role id:" + roleId + " apiName: " + apiName);
+        }
+    }
+
+    private void addRoleColumnAndMigrateAccountTable(final Connection conn, final RoleType[] roleTypes) {
+        final String alterTableSql = "ALTER TABLE `cloud`.`account` ADD COLUMN `role_id` bigint(20) unsigned COMMENT 'role id for this account' AFTER `type`, " +
+                "ADD KEY `fk_account__role_id` (`role_id`), " +
+                "ADD CONSTRAINT `fk_account__role_id` FOREIGN KEY (`role_id`) REFERENCES `roles` (`id`);";
+        try (PreparedStatement pstmt = conn.prepareStatement(alterTableSql)) {
+            pstmt.executeUpdate();
+            s_logger.info("Altered cloud.account table and added column role_id");
+        } catch (SQLException e) {
+            if (e.getMessage().contains("role_id")) {
+                s_logger.warn("cloud.account table already has the role_id column, skipping altering table and migration of accounts");
+                return;
+            } else {
+                throw new CloudRuntimeException("Unable to create column quota_calculated in table cloud_usage.cloud_usage", e);
+            }
+        }
+        migrateAccountsToDefaultRoles(conn, roleTypes);
+    }
+
+    private void migrateAccountsToDefaultRoles(final Connection conn, final RoleType[] roleTypes) {
+        try (PreparedStatement selectStatement = conn.prepareStatement("SELECT `id`, `type` FROM `cloud`.`account`;");
+             ResultSet selectResultSet = selectStatement.executeQuery()) {
+            while (selectResultSet.next()) {
+                Long accountId = selectResultSet.getLong(1);
+                Short accountType = selectResultSet.getShort(2);
+                Long roleId = null;
+                for (RoleType roleType : roleTypes) {
+                    if (roleType.getAccountType() == accountType) {
+                        roleId = roleType.getId();
+                        break;
+                    }
+                }
+                if (roleId == null) {
+                    continue;
+                }
+                try (PreparedStatement updateStatement = conn.prepareStatement("UPDATE `cloud`.`account` SET role_id = ? WHERE id = ?;")) {
+                    updateStatement.setLong(1, roleId);
+                    updateStatement.setLong(2, accountId);
+                    updateStatement.executeUpdate();
+                    updateStatement.close();
+
+                } catch (SQLException e) {
+                    s_logger.error("Failed to update cloud.account role_id for account id:" + accountId + " with exception: " + e.getMessage());
+                    throw new CloudRuntimeException("Exception while updating cloud.account role_id", e);
+                }
+            }
```

Comment:
at the time of upgrade there will be only a single thread running this, performing the upgrades so no explicit need of a transaction. The exception is thrown so that it can appear to the sysadmin as something needing manual fixing.